base/strings/substring.jl

```diff
@@ -162,10 +139,15 @@ julia> join(reverse(collect(graphemes("ax̂e")))) # reverses graphemes
 "ex̂a"
 ```
 """
-reverse(s::AbstractString) = RevString(s)
-reverse(s::RevString) = s.string
-
-## reverse an index i so that reverse(s)[i] == s[reverseind(s,i)]
+function reverse(s::AbstractString)::String
```

Comment:
I thought in #24613 you wanted to remove the `reverse(::AbstractString)` fallback, and always return a string of the same encoding (so that `reverseind` works)?

So shouldn't this be `s::Union{String,SubString{String}}`, here?

Reply:
Yes, I should definitely fix this. This is a holdover from the first pass through this when I thought that we could have `reverse` always produce a `String`.